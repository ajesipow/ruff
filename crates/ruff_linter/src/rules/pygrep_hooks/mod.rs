//! Rules from [pygrep-hooks](https://github.com/pre-commit/pygrep-hooks).
pub(crate) mod rules;

#[cfg(test)]
mod tests {
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::registry::Rule;

    use crate::settings::types::PreviewMode;
    use crate::test::test_path;
    use crate::{assert_messages, settings};

    #[test_case(
        Rule::BlanketTypeIgnore,
        Path::new("PGH003_0.py"),
        PreviewMode::Disabled
    )]
    #[test_case(
        Rule::BlanketTypeIgnore,
        Path::new("PGH003_1.py"),
        PreviewMode::Disabled
    )]
    #[test_case(Rule::BlanketNOQA, Path::new("PGH004_0.py"), PreviewMode::Disabled)]
    #[test_case(Rule::BlanketNOQA, Path::new("PGH004_1.py"), PreviewMode::Disabled)]
    #[test_case(Rule::BlanketNOQA, Path::new("PGH004_2.py"), PreviewMode::Disabled)]
    #[test_case(Rule::BlanketNOQA, Path::new("PGH004_2.py"), PreviewMode::Enabled)]
    #[test_case(Rule::BlanketNOQA, Path::new("PGH004_3.py"), PreviewMode::Disabled)]
    #[test_case(Rule::BlanketNOQA, Path::new("PGH004_3.py"), PreviewMode::Enabled)]
    #[test_case(
        Rule::InvalidMockAccess,
        Path::new("PGH005_0.py"),
        PreviewMode::Disabled
    )]
    fn rules(rule_code: Rule, path: &Path, preview: PreviewMode) -> Result<()> {
        let snapshot = {
            let base = format!("{}_{}", rule_code.noqa_code(), path.to_string_lossy());
            // To keep snapshot filenames more succinct we're only adding the preview suffix when
            // enabled.
            match preview {
                PreviewMode::Disabled => base,
                PreviewMode::Enabled => {
                    format!("{base}_preview_{preview}")
                }
            }
        };
        let mut settings = settings::LinterSettings::for_rule(rule_code);
        settings.preview = preview;
        let diagnostics = test_path(Path::new("pygrep_hooks").join(path).as_path(), &settings)?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
