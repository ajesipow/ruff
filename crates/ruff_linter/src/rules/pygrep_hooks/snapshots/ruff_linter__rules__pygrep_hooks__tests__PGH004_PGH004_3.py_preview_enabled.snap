---
source: crates/ruff_linter/src/rules/pygrep_hooks/mod.rs
---
PGH004_3.py:1:1: PGH004 Use specific rule codes when using `noqa`
  |
1 | # flake8: noqa
  | ^^^^^^^^^^^^^^ PGH004
  |
